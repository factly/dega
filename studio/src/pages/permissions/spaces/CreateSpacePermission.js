import React from 'react';
import SpacePermissionCreateForm from './components/PermissionForm';
import { useDispatch } from 'react-redux';
import { addSpacePermission } from '../../../actions/spacePermissions';

import useNavigation from '../../../utils/useNavigation';

function CreateSpacePermission() {
  const history = useNavigation();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpacePermission(values)).then(() => history('/admin/permissions/spaces'));
  };
  return <SpacePermissionCreateForm onCreate={onCreate} />;
}

export default CreateSpacePermission;
