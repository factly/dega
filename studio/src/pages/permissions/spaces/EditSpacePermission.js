import React from 'react';
import SpacePermissionEditForm from './components/PermissionForm';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';
import { updateSpacePermission, getSpaces } from '../../../actions/spacePermissions';

import { useParams } from 'react-router-dom';
import RecordNotFound from '../../../components/ErrorsAndImage/RecordNotFound';
import useNavigation from '../../../utils/useNavigation';

function EditSpacePermission() {
  const history = useNavigation();
  const { sid, pid } = useParams();

  const dispatch = useDispatch();

  const { space, loading } = useSelector((state) => {
    return {
      space: state.spacePermissions.details[sid] ? state.spacePermissions.details[sid] : null,
      loading: state.spaces.loading,
    };
  });

  React.useEffect(() => {
    dispatch(getSpaces());
  }, [dispatch, sid, pid]);

  if (!space) {
    return <RecordNotFound />;
  }

  const onUpdate = (values) => {
    dispatch(updateSpacePermission({ ...space.permission, ...values })).then(() =>
      history('/admin/permissions/spaces'),
    );
  };
  if (space) {
    return <SpacePermissionEditForm data={space.permission} onCreate={onUpdate} />;
  }
  if (loading) return <Skeleton />;
}

export default EditSpacePermission;
