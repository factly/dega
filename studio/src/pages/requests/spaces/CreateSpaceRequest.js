import React from 'react';
import SpaceRequestCreateForm from './components/RequestForm';
import { useDispatch } from 'react-redux';
import { addSpaceRequest } from '../../../actions/spaceRequests';
 
import { Helmet } from 'react-helmet';
import useNavigation from '../../../utils/useNavigation';

function CreateSpaceRequest() {
  const history = useNavigation();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpaceRequest(values)).then(() => history('/admin/requests/spaces'));
  };
  return (
    <>
      <Helmet title={'Create Space Request'} />
      <SpaceRequestCreateForm onCreate={onCreate} />
    </>
  );
}

export default CreateSpaceRequest;
