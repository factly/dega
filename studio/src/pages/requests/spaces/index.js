import React from 'react';
import SpaceRequestList from './components/RequestList';
import { Space, Button, Row } from 'antd';
import { Link } from 'react-router-dom';
import { PlusOutlined } from '@ant-design/icons';
import { Helmet } from 'react-helmet';

function SpaceRequests() {
  return (
    <Space direction="vertical">
      <Helmet title={'Space Requests'} />
      <Row gutter={16} justify="end">
        <Link key="1" to="/admin/requests/spaces/create">
          <Button
            icon={<PlusOutlined />}
            type="primary">Create</Button>
        </Link>
      </Row>

      <SpaceRequestList />
    </Space>
  );
}

export default SpaceRequests;
