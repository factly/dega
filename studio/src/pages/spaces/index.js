import React from 'react';
import { Space, Button, Row, Col, ConfigProvider } from 'antd';
import { Link } from 'react-router-dom';
import SpaceList from './components/SpaceList';
import { useSelector } from 'react-redux';
import { Helmet } from 'react-helmet';

function Spaces() {
  const { role } = useSelector((state) => {
    const { selected } = state.spaces;

    if (selected !== '') {
      const space = state.spaces.details[selected];
      const role = space.org_role;
      return {
        role: role,
      };
    }
    return { role: 'member' };
  });

  return (
    <ConfigProvider
      theme={{
        components: {
          Button: {
            controlHeight: 35,
            colorBorder: '#1890FF',
            colorPrimaryHover: '#00000026',
            colorText: '#1890FF',
            paddingXS: '2px, 16px, 2px, 16px',
          },
        },
      }}
    >
      <Space direction="vertical">
        <Helmet title={'Spaces'} />
        <Row gutter={16} justify="end">
          {role === 'admin' ? (
            <Col>
              <Link key="2" to="/settings/advanced/reindex">
                <Button>Reindex</Button>
              </Link>
            </Col>
          ) : null}
          <Link key="1" to="/spaces/create">
            <Button type="primary">New Space</Button>
          </Link>
        </Row>
        <SpaceList />
      </Space>
    </ConfigProvider>
  );
}

export default Spaces;
