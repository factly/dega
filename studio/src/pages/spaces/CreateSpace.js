import React from 'react';
import SpaceCreateForm from './components/SpaceCreateForm';
import { useDispatch } from 'react-redux';
import { addSpace, getSpaces } from '../../actions/spaces';

import { Helmet } from 'react-helmet';
import useNavigation from '../../utils/useNavigation';

function CreateSpace() {
  const history = useNavigation();

  const dispatch = useDispatch();
  const onCreate = (values) => {
    dispatch(addSpace(values)).then(() => {
      dispatch(getSpaces());
      history('/admin/spaces');
    });
  };
  return (
    <>
      <Helmet title={'Create Space'} />
      <SpaceCreateForm onCreate={onCreate} />
    </>
  );
}

export default CreateSpace;
