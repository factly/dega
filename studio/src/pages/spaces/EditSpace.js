import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Skeleton } from 'antd';
import { updateSpace } from '../../actions/spaces';
import SpaceEditForm from './components/SpaceEditForm';
import RecordNotFound from '../../components/ErrorsAndImage/RecordNotFound';
import { Helmet } from 'react-helmet';
import useNavigation from '../../utils/useNavigation';

function EditSpace() {
  const history = useNavigation();
  const { id } = useParams();
  const dispatch = useDispatch();

  const { space, loading } = useSelector((state) => {
    return {
      space: state.spaces.details[id],
      loading: state.spaces.loading,
    };
  });

  const onCreate = (values) => {
    dispatch(updateSpace({ ...space, ...values })).then(() => history(`/admin/spaces`));
  };

  if (loading) return <Skeleton />;

  if (!space) {
    return <RecordNotFound />;
  }

  return (
    <>
      <Helmet title={`${space?.name} - Edit Space`} />
      <SpaceEditForm onCreate={onCreate} data={space} />
    </>
  );
}

export default EditSpace;
