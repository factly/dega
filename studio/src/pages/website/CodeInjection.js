import React, { useState } from 'react';
import { Button, Form, Skeleton, Row } from 'antd';
import MonacoEditor from '../../components/MonacoEditor/index';
import { useSelector, useDispatch } from 'react-redux';
import { updateSpace } from '../../actions/spaces';
import RecordNotFound from '../../components/ErrorsAndImage/RecordNotFound';
import { Helmet } from 'react-helmet';

function CodeInjection() {
  const id = useSelector((state) => state.spaces.selected);
  const dispatch = useDispatch();
  const [form] = Form.useForm();
  const [valueChange, setValueChange] = useState(false);

  const { space, loading } = useSelector((state) => {
    return {
      space: state.spaces.details[id],
      loading: state.spaces.loading,
    };
  });

  const onCreate = (values) => {
    dispatch(updateSpace({ ...space, ...values }));
  };

  if (loading) return <Skeleton />;

  if (!space) {
    return <RecordNotFound />;
  }

 

  const onReset = () => {
    form.resetFields();
  };

  return (
    <div>
      <Helmet title={'Code Injection'} />
      <Form
        form={form}
        layout="vertical"
        initialValues={space}
        name="code-injection"
        onFinish={(values) => {
          onCreate(values);
          // onReset();
        }}
        scrollToFirstError={true}
        onFinishFailed={(errors) => {
          //let name = errors.errorFields[0].name[0];
          // if (['name', 'slug'].includes(name)) {
          console.log({ errors });
          // }
        }}
        onValuesChange={() => {
          setValueChange(true);
        }}
      >
        <Row justify="end">
          <Form.Item>
            <Button disabled={!valueChange} type="primary" htmlType="submit">
              Update
            </Button>
          </Form.Item>
        </Row>
        <Form.Item name="header_code" label="Header Code">
          <MonacoEditor language="html" width="100%" />
        </Form.Item>
        <Form.Item name="footer_code" label="Footer Code">
          <MonacoEditor language="html" width="100%" />
        </Form.Item>
      </Form>
    </div>
  );
}

export default CodeInjection;
