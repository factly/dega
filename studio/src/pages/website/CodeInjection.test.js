import React from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch, Provider, useSelector } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount, shallow } from 'enzyme';
import { act } from '@testing-library/react';

import '../../matchMedia.mock';

import CodeInjection from './CodeInjection';
import * as actions from '../../actions/spaces';
import { Form } from 'antd';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('react-monaco-editor', () => {
  const MonacoEditor = () => <div />;
  return MonacoEditor;
});
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));
jest.mock('../../actions/spaces', () => ({
  updateSpace: jest.fn(),
}));

describe('CodeInjection component', () => {
  let store;
  store = mockStore({
    spaces: {
      orgs: [
        {
          id: 34,
          created_at: '2022-04-29T07:10:05.626434Z',
          updated_at: '2022-04-29T07:10:05.626434Z',
          deleted_at: null,
          created_by_id: 34,
          updated_by_id: 34,
          title: 'test',
          slug: 'test',
          permission: {
            id: 34,
            created_at: '2022-04-29T07:10:05.637767Z',
            updated_at: '2022-04-29T07:10:05.637767Z',
            deleted_at: null,
            created_by_id: 34,
            updated_by_id: 34,
            role: 'owner',
          },
          spaces: [2],
        },
      ],
      details: {
        '2': {
          id: 2,
          created_at: '2022-04-29T11:19:00.634414Z',
          updated_at: '2022-06-28T08:58:44.782829Z',
          deleted_at: null,
          created_by_id: 34,
          updated_by_id: 34,
          name: 'factly-eng',
          slug: 'factly-eng',
          site_title: '',
          tag_line: '',
          description: '',
          site_address: '',
          header_code: '',
          footer_code: '',
          contact_info: null,
          organisation_id: 34,
        },
      },
      loading: false,
      selected: 2,
    },
  });
  store.dispatch = jest.fn(() => ({}));
  const mockedDispatch = jest.fn(() => Promise.resolve({}));
  useDispatch.mockReturnValue(mockedDispatch);
  describe('snapshot testing', () => {
    it('should render the component', () => {
      let component;
      act(() => {
        component = shallow(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
      expect(component).toMatchSnapshot();
      act(() => component.unmount());
    });
  });
  it('should match skeleton while loading', () => {
    let tree;
    store = mockStore({
      spaces: {
        orgs: [],
        details: {},
        loading: true,
        selected: 0,
      },
    });
    act(() => {
      tree = shallow(
        <Provider store={store}>
          <CodeInjection />
        </Provider>,
      );
    });
    expect(tree).toMatchSnapshot();
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      store = mockStore({
        spaces: {
          orgs: [
            {
              id: 34,
              created_at: '2022-04-29T07:10:05.626434Z',
              updated_at: '2022-04-29T07:10:05.626434Z',
              deleted_at: null,
              created_by_id: 34,
              updated_by_id: 34,
              title: 'test',
              slug: 'test',
              permission: {
                id: 34,
                created_at: '2022-04-29T07:10:05.637767Z',
                updated_at: '2022-04-29T07:10:05.637767Z',
                deleted_at: null,
                created_by_id: 34,
                updated_by_id: 34,
                role: 'owner',
              },
              spaces: [2],
            },
          ],
          details: {
            '2': {
              id: 2,
              created_at: '2022-04-29T11:19:00.634414Z',
              updated_at: '2022-06-28T08:58:44.782829Z',
              deleted_at: null,
              created_by_id: 34,
              updated_by_id: 34,
              name: 'factly-eng',
              slug: 'factly-eng',
              site_title: '',
              tag_line: '',
              description: '',
              site_address: '',
              header_code: '',
              footer_code: '',
              contact_info: null,
              organisation_id: 34,
            },
          },
          loading: false,
          selected: 2,
        },
      });
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should display RecordNotFound when no space found', () => {
      let store = mockStore({
        spaces: {
          orgs: [],
          details: {},
          loading: false,
          selected: 0,
        },
        media: {
          req: [],
          details: {},
          loading: true,
        },
      });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
      expect(wrapper.find('RecordNotFound').length).toBe(1);
      expect(wrapper.find(Form).length).toBe(0);
    });

    it('should call updateSpace', (done) => {
      actions.updateSpace.mockReset();
      useDispatch.mockReturnValueOnce(() => Promise.resolve({}));
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CodeInjection />
          </Provider>,
        );
      });
      act(() => {
        wrapper
          .find('MonacoEditor')
          .at(0)
          .props()
          .onChange({ target: { value: 'header-code' } });
        wrapper
          .find('MonacoEditor')
          .at(1)
          .props()
          .onChange({ target: { value: 'footer-code' } });
        wrapper.find('Button').at(0).simulate('submit');
      });
      setTimeout(() => {
        expect(actions.updateSpace).toHaveBeenCalledWith({
          id: 2,
          created_at: '2022-04-29T11:19:00.634414Z',
          updated_at: '2022-06-28T08:58:44.782829Z',
          deleted_at: null,
          created_by_id: 34,
          updated_by_id: 34,
          organisation_id: 34,
          name: 'factly-eng',
          slug: 'factly-eng',
          footer_code: 'footer-code',
          header_code: 'header-code',
          site_title: '',
          tag_line: '',
          description: '',
          site_address: '',
          contact_info: null,
        });
        done();
      }, 0);
    });
  });
});
