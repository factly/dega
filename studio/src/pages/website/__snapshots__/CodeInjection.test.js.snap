// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeInjection component should match skeleton while loading 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <CodeInjection />
</ContextProvider>
`;

exports[`CodeInjection component snapshot testing should render the component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <CodeInjection />
</ContextProvider>
`;
